src/values/AbstractObjectValue.js

```diff
@@ -113,6 +113,13 @@ export default class AbstractObjectValue extends AbstractValue {
   }
 
   makeSimple(): void {
+    if (this.values.isTop() && this.getType() === ObjectValue) {
+      let obj = new ObjectValue(this.$Realm, this.$Realm.intrinsics.ObjectPrototype);
+      obj.intrinsicName = this.intrinsicName;
+      obj.intrinsicNameGenerated = true;
+      obj.makePartial();
```

Comment:
You could add `.refuseSerialization = true;` to make this thing more obscure.

Reply:
Actually, we DO serialize template objects, hence to assignments to obj.intrinsicName*. The reason being that we get properties from the template objects and these properties become temporal (derived) values because the root object is intrinsic. It is the right behavior, but it does mean that this is not proxy for "template did not leak".